client/my-sites/stats/wordads-chart-tabs/index.jsx

```diff
@@ -0,0 +1,256 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { find, flowRight } from 'lodash';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import ElementChart from 'components/chart';
+import Legend from 'components/chart/legend';
+import StatTabs from '../stats-tabs';
+import StatsModulePlaceholder from '../stats-module/placeholder';
+import Card from 'components/card';
+import QuerySiteStats from 'components/data/query-site-stats';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsNormalizedData,
+} from 'state/stats/lists/selectors';
+import { recordGoogleEvent } from 'state/analytics/actions';
+import { rangeOfPeriod } from 'state/stats/lists/utils';
+
+class WordAdsChartTabs extends Component {
+	constructor( props ) {
+		super( props );
+		const activeTab = this.getActiveTab();
+		const activeCharts = activeTab.legendOptions ? activeTab.legendOptions.slice() : [];
+		this.state = {
+			activeLegendCharts: activeCharts,
+			activeTab: activeTab,
+		};
+	}
+
+	UNSAFE_componentWillReceiveProps( nextProps ) {
```

Comment:
can we use `getDerivedStateFromProps` or memoization instead? I'd rather not add more UNSAFE_ methods and have to port them later. See https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html